src/compas_fab/robots/robot.py

```diff
@@ -1125,25 +1159,27 @@ def send_trajectory(self):
         self.ensure_client()
         raise NotImplementedError
 
-    @property
-    def frames(self):
-        """The robot's frames."""
-        return self.model.frames
+    def updated_frames(self, configuration, group=None):
+        """Returns the robot's updated frames."""
+        joint_names = self.get_configurable_joint_names(group)
+        joint_state = dict(zip(joint_names, configuration.values))
+        return self.model.updated_frames(joint_state)
 
-    @property
-    def axes(self):
-        """The robot's axes."""
-        return self.model.axes
+    def updated_axes(self, configuration, group=None):
+        """Returns the robot's updated axes."""
+        joint_names = self.get_configurable_joint_names(group)
+        joint_state = dict(zip(joint_names, configuration.values))
+        return self.model.updated_axes(joint_state)
 
     # ==========================================================================
     # drawing
     # ==========================================================================
 
-    def update(self, configuration, group=None, visual=True, collision=True):
+    def update(self, configuration, group=None, collision=True):
         """Updates the robot's geometry.
         """
         names = self.get_configurable_joint_names(group)
-        self.artist.update(configuration, names, visual, collision)
+        self.artist.update(configuration, names, collision)
```

Comment:
Wouldn't it be actually desirable to be able to not transform visual geometry very often?

Reply:
added parameter visual